app/policies/admin_setting_policy.rb

```diff
@@ -0,0 +1,26 @@
+class AdminSettingPolicy < ApplicationPolicy
+  SETTINGS_ROLES = %w(superadmin tag_wrangling).freeze
+
+  def index?
+    update?
+  end
+
+  def update?
+    user_has_roles?(SETTINGS_ROLES)
+  end
+
+  def permitted_attributes
+    if user.roles.include?('superadmin')
+      [
+        :account_creation_enabled, :invite_from_queue_enabled, :invite_from_queue_number,
+        :invite_from_queue_frequency, :days_to_purge_unactivated, :last_updated_by,
```

Comment:
This is an existing issue, but we could fix it here: can you remove `last_updated_by` from the list of permitted attributes? As a result, you'll have to remove the hidden field `last_updated_by` from the settings form, then set `last_updated_by` to `current_admin` in the settings controller.